sdk/core/azure-core-amqp/inc/azure/core/amqp/message_sender.hpp

```diff
@@ -107,38 +109,8 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     using MessageSendCompleteCallback
         = std::function<void(MessageSendStatus sendResult, Models::AmqpValue const& deliveryState)>;
 
-    /** @brief Construct a message sender.
-     *
-     * @param session The AMQP session.
-     * @param target The target to which messages will be sent.
-     * @param options The options to use when sending messages.
-     * @param events The events handler for the message sender.
-     *
-     */
-    MessageSender(
-        Session const& session,
-        Models::_internal::MessageTarget const& target,
-        MessageSenderOptions const& options,
-        MessageSenderEvents* events);
-
-    /** @brief Construct a message sender for use in a message receiving handler.
-     *
-     * @param session The AMQP session.
-     * @param newLinkEndpoint The link endpoint to use for sending messages.
-     * @param target The target to which messages will be sent.
-     * @param options The options to use when sending messages.
-     * @param events The events handler for the message sender.
-     *
-     * @remarks This constructor is primarily used for testing scenarios.
-     */
-    MessageSender(
-        Session const& session,
-        LinkEndpoint& newLinkEndpoint,
-        Models::_internal::MessageTarget const& target,
-        MessageSenderOptions const& options,
-        MessageSenderEvents* events);
-
     ~MessageSender() noexcept;
+    operator bool() const noexcept { return m_impl != nullptr; }
```

Comment:
I think `IsInitialized()` could be better than implicit operator bool.
Operators are mostly for simulating built-in type semantics, so that it looks natural, and for templates to work with both built-in types and the user-defined type, but here, without reading documentation, you ask me what operator bool does, I don't think I could always guess correctly.
```suggestion
    bool IsInitialized() const noexcept { return m_impl != nullptr; }
```

Reply:
This is a tough one. I'm a fan of using `operator bool` as a mechanism to indicate validity (similar to what we do for `Azure::Nullable` and the STL does for `std::optional`. 

In the case of message sender and receiver, the operator bool is uninteresting (it was when they had default constructors, but that is no longer the case), so I've removed it.

But I'm not nearly as comfortable with removing it from `AmqpError` and `AmqpMessage` because it allows for what I believe is a more natural syntax.